src/pudl/cli.py

```diff
@@ -86,14 +87,23 @@ def main():
         logfile=args.logfile, loglevel=args.loglevel
     )
 
+    etl_settings = EtlSettings.from_yaml(args.settings_file)
+
+    if (not os.getenv("PUDL_OUT")) or (not os.getenv("PUDL_CACHE")):
+        pudl_settings = pudl.workspace.setup.derive_paths(
+            pudl_in=etl_settings.pudl_in, pudl_out=etl_settings.pudl_out
+        )
+
+        os.environ["PUDL_CACHE"] = pudl_settings["data_dir"]
+        os.environ["PUDL_OUTPUT"] = pudl_settings["pudl_out"]
+        os.environ["DAGSTER_HOME"] = pudl_settings["pudl_in"]
+
```

Comment:
We might want to add a comment or a deprecation warning here that explains why we're falling back to `pudl_settings` if the env vars aren't set.